### PR TITLE
Add Solana Signing Support, Examples

### PR DESCRIPTION
**Pull request by ProfMoo**
### What changed? Why?

This PR adds an abstraction around the SDK's signing support in order to support signing staking operations for ETH and SOL using the same interface. This PR relies heavily on Golang's stdlib `crypto/rand`.

Alongside these changes, I included numerous Solana stake examples to provide full examples of how to stake SOL.

Our current Solana staking endpoint can take significant periods of time to get data from the underlying Solana nodes, so I've added the ability to modify the timeout on the client.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: examples/solana/devnet/claim-stake/main.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	if err != nil {
+		return "", fmt.Errorf("error marshaling transaction: %w", err)
+	}
+
+	base58EncodedSignedTx := base58.Encode(marshaledTx)
+
+	return base58EncodedSignedTx, nil
+}
+
+func broadcastSolTransaction(ctx context.Context, signedTx string) (string, error) {
```
Comment (drohit-cb) [inline examples/solana/devnet/claim-stake/main.go:130]: I really think we should not be duplicating the sign/broadcast for every variation.

Comment (ProfMoo) [reply examples/solana/devnet/claim-stake/main.go:130]: I agree. Let me see if we can incorporate this into the SDK to help standardize this functionality.

File: examples/solana/build-staking-operation/main.go
Comment (ProfMoo) [inline examples/solana/build-staking-operation/main.go:1]: This example was actually doing numerous thing besides just `build-staking-operation`. I've split them up into 3 examples.

**Review by deangalvin-cb — COMMENTED** [review]
This looks great overall! I think a few unit tests & this is complete!

**Comment by ProfMoo** [comment]
@deangalvin-cb I've added a couple unit test suites for the `Signable` implementations and updated the existing `StakingOperation` suite.

**Review by deangalvin-cb — APPROVED** [review]
Love this pattern!

**Comment by drohit-cb** [comment]
@ProfMoo Love this. Thanks! Would be nice to follow up on the NodeJS SDK as well and have the same pattern both places.

**Pull request merged**